src/main/kotlin/com/katanox/tabour/integration/sqs/core/consumer/SqsEventPoller.kt

```diff
@@ -17,69 +32,96 @@ import java.util.concurrent.TimeUnit
 private val logger = KotlinLogging.logger {}
 
 class SqsEventPoller(
-    private val name: String,
+    private val queueUrl: String,
     private val eventHandler: SqsEventHandler,
-    private val eventFetcher: SqsEventFetcher,
-    private val pollerThreadPool: ScheduledThreadPoolExecutor,
-    private val handlerThreadPool: ThreadPoolExecutor,
-    private val pollingProperties: EventPollerProperties,
-    private val sqsConfiguration: SqsConfiguration,
-    private val exceptionHandler: ExceptionHandler
+    private val client: SqsClient,
+    private val pollerConfigs: EventPollerProperties,
+    private val tabourConfigs: TabourAutoConfigs,
 ) {
```

Comment:
based on my previous comment, i would add here the context stuff:

```kotlin
class SqsEventPoller(...): CoroutineScope {
    private val supervisorJob = SupervisorJob()
    override val coroutineContext: CoroutineContext
        get() = Dispatchers.IO + supervisorJob
}
```

Supervisor job will also prevent coroutines crashing each other if something goes wrong